lib/omniauth/strategies/mlh.rb

```diff
@@ -5,42 +5,103 @@
+    # @example Basic Usage
+    #   use OmniAuth::Builder do
+    #     provider :mlh, ENV['MLH_KEY'], ENV['MLH_SECRET']
+    #   end
+    #
+    # @example With Expandable Fields
+    #   use OmniAuth::Builder do
+    #     provider :mlh, ENV['MLH_KEY'], ENV['MLH_SECRET'],
+    #              expand_fields: ['education', 'professional_experience']
+    #   end
+    #
+    # @example With Refresh Tokens (offline access)
+    #   use OmniAuth::Builder do
+    #     provider :mlh, ENV['MLH_KEY'], ENV['MLH_SECRET'],
+    #              scope: 'user:read:profile offline_access'
+    #   end
+    #
+    # When offline_access scope is requested, the strategy will include
+    # refresh_token in the credentials hash if provided by the server.
     class MLH < OmniAuth::Strategies::OAuth2 # :nodoc:
       option :name, :mlh
 
       option :client_options, {
         site: 'https://my.mlh.io',
         authorize_url: 'oauth/authorize',
-        token_url: 'oauth/token'
+        token_url: 'oauth/token',
+        auth_scheme: :request_body # Change from basic auth to request body
       }
 
+      # Support expandable fields through options
+      option :expand_fields, []
+
       uid { data[:id] }
 
       info do
-        data.slice(
-          :email,
-          :created_at,
-          :updated_at,
-          :first_name,
-          :last_name,
-          :level_of_study,
-          :major,
-          :date_of_birth,
-          :gender,
-          :phone_number,
-          :profession_type,
-          :company_name,
-          :company_title,
-          :scopes,
-          :school
-        )
+        prune_hash({
+                     # Basic fields
+                     id: data[:id],
+                     created_at: data[:created_at],
+                     updated_at: data[:updated_at],
+                     first_name: data[:first_name],
+                     last_name: data[:last_name],
+                     email: data[:email],
+                     phone_number: data[:phone_number],
+                     roles: data[:roles],
+
+                     # Expandable fields
+                     profile: data[:profile],
+                     address: data[:address],
+                     social_profiles: data[:social_profiles],
+                     professional_experience: data[:professional_experience],
+                     education: data[:education],
+                     identifiers: data[:identifiers]
+                   })
       end
 
       def data
-        @data ||= begin
-          access_token.get('/api/v3/user.json').parsed.deep_symbolize_keys[:data]
-        rescue StandardError
-          {}
+        @data ||= fetch_and_process_data
+      rescue StandardError
+        {}
+      end
+
+      private
+
+      def fetch_and_process_data
+        response = access_token.get(build_api_url)
+        data = JSON.parse(response.body, symbolize_names: true)
+        return {} unless data.is_a?(Hash)
+
+        symbolize_nested_arrays(data)
```

Comment:
omniauth-oauth2 (the gem that omniauth-mlh is built on top of) uses something weird called SnakyHash instead of standard Ruby hashes. It was breaking deep_symbolize_keys so I changed it to this